OpenRA.Mods.Common/Traits/AttackMove.cs

```diff
@@ -130,16 +142,24 @@ public override void Tick(World world)
 
 		public override string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi)
 		{
-			var prefix = mi.Modifiers.HasModifier(Modifiers.Ctrl) ? "assaultmove" : "attackmove";
-
 			if (world.Map.Contains(cell) && subjects.Any())
 			{
-				var explored = subjects.First().Actor.Owner.Shroud.IsExplored(cell);
+				var subject = subjects.First();
+				var explored = subject.Actor.Owner.Shroud.IsExplored(cell);
 				var blocked = !explored && subjects.Any(a => !a.Trait.Info.MoveIntoShroud);
-				return blocked ? prefix + "-blocked" : prefix;
+				var info = subject.Trait.Info;
+				var prefix = mi.Modifiers.HasModifier(Modifiers.Ctrl);
+				if (prefix && !blocked)
+					return info.AssaultMoveCursor;
+				if (prefix && blocked)
+					return info.AssaultMoveBlockedCursor;
+				if (prefix && !blocked)
```

Comment:
These are the same as the checks above, so will never run.

Reply:
That was sloppy.